src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/DataTotal.java

```diff
@@ -3,19 +3,23 @@
 import org.apache.commons.lang3.StringUtils;
 
 import java.math.BigDecimal;
+import java.util.List;
 import java.util.Map;
+import java.util.function.BiFunction;
 
 public class DataTotal {
 
-    public BigDecimal calculate(Map<String,String> dataMap, Map<String,String> addFields, Map<String,String> subFields){
-        BigDecimal total = BigDecimal.ZERO;
-        for(Map.Entry<String, String> field : addFields.entrySet()) {
-            total = total.add(parseCurrency(dataMap, field.getKey(), field.getValue()));
-        }
-        for(Map.Entry<String, String> field: subFields.entrySet()) {
-            total = total.subtract(parseCurrency(dataMap, field.getKey(), field.getValue()));
-        }
-        return total;
+    BiFunction<Map<String,String>, Map<String, List<String>>, BigDecimal> sumFields = (dataMap, fields) -> {
```

Comment:
Really like the use of BiFunction here, for some people though this might be a little complicated (maybe worth adding a quick comment here to explain a little.

Reply:
I've added a little comment here